projects/systelab-components/src/lib/systelab-components.module.ts

```diff
@@ -294,4 +298,21 @@ import { NumpadDecimalNumericDirective } from './directives/numpad-decimal-numer
 	]
 })
 export class SystelabComponentsModule {
+
+	constructor(@Optional() @SkipSelf() parentModule: SystelabComponentsModule) {
+		if(parentModule) {
+			throw new Error('SystelabComponentsModule is already loaded. Please add it in AppModule only.');
+		}
+	}
+
+	public static forRoot(conf?: AppConfig): ModuleWithProviders<SystelabComponentsModule> {
+		return {
+			ngModule: SystelabComponentsModule,
+			providers: [
+				{
+					provide: APP_CONFIG, useValue: conf
```

Comment:
What if configuration is not defined? Shouldn't we provide a default one?
I would suggest to set "productionMode" to true by default.